hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -641,6 +661,36 @@ func (r *HostedClusterReconciler) reconcileControlPlaneOperator(ctx context.Cont
 	return nil
 }
 
+func (r *HostedClusterReconciler) reconcileIgnitionServer(ctx context.Context, hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) error {
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(controlPlaneNamespace), controlPlaneNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane namespace: %w", err)
+	}
+
+	// Reconcile service
+	// TODO (alberto): enable nodePort choice at the hostedClusterAPI
+	ignitionServerService := ignitionserver.Service(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, ignitionServerService, func() error {
+		ignitionserver.ReconcileServiceClusterIP(ignitionServerService)
```

Comment:
For consistency with the rest of this code, for each of these (ReconcileServiceClusterIP, ReconcileRoute, ReconcileDeployment), can you move the reconcile code inline to this function? And if they're big enough to warrant testing or something, make them stateless functions inside this file (ReconcileDeployment would be the only one I see that may qualify, but it has no real behavior other than a straight up transformation so not sure if even that's worth its own function).